### DISCUSSION TITLE
Vercel Staging Environment

### DISCUSSION THREAD

**Discussion by votemike** [vercel/vercel#10770] [Help]
### Question

I can easily create a production environment with production code and production data sources.
Vercel give us preview environments with staged code and our choice of data sources.
However I cannot figure out a way to have an environment with production code and pre-production data sources.

I could create a staging domain with a Git Branch of "main". But there is no way I can see to assign pre-production env variables to this domain.

My end goal is to allow users to edit data sources or config to data sources and see what effect that would have on the code that is currently in main.

**Comment by mikedidomizio** [comment]
Maybe what you're asking is more complicated, but if you're fine changing the workflow and the staging domain points to an intermediate staging branch, [this may work for you](https://vercel.com/guides/set-up-a-staging-environment-on-vercel).

Your workflow will change so that production level code will be merged into the staging branch, where it can have custom env vars between preview and production.  Then when satisfied, you merge the staging branch code into production, which deploys with your production env vars.

**Comment by votemike** [comment]
I have spoken to Vercel about this.
At the moment the option closest to my desires is to create a 2nd project pointing at the same repo. Add different env vars to this project.
Then upon merge to main, both projects will deploy. One to my production site. One to my staging site.

Although I will leave this discussion open in the hope that a better solution comes around in the future.
